pcdsdevices/targets.py

```diff
@@ -260,3 +268,907 @@ def move(self, nxspaces, nyspaces, wait=False):
+        """
+        path = path or self._path
+        with open(path, 'w') as sample_file:
+            yaml.safe_dump(None, sample_file)
+
+    def map_points(self, snake_like=True, top_left=None, top_right=None,
+                   bottom_right=None, bottom_left=None, m_rows=None,
+                   n_columns=None):
+        """
+        Map the points of a quadrilateral.
+
+        Given the 4 corners coordinates of a grid, and the numbers of rows and
+        columns, map all the sample positions in 2-d coordinates.
+
+        Parameters
+        ----------
+        snake_like : bool
+            Indicates if the points should be saved in a snake_like pattern.
+        top_left : tuple, optional
+            (x, y) coordinates of the top left corner
+        top_right : tuple, optional
+            (x, y) coordinates of the top right corner
+        bottom_right : tuple, optional
+            (x, y) coordinates of the bottom right corner
+        bottom_left : tuple, optional
+            (x, y) coordinates of the bottom left corner
+        m_rows : int, optional
+            Number of rows the grid has.
+        n_columns : int, optional
+            Number of columns the grid has.
+
+        Returns
+        -------
+        xx, yy : tuple
+            Tuple of two lists with all mapped points for x and y positions in
+            the grid.
+        """
+        top_left = top_left or self.get_presets()[0]
+        top_right = top_right or self.get_presets()[1]
+        bottom_right = bottom_right or self.get_presets()[2]
+        bottom_left = bottom_left or self.get_presets()[3]
+
+        if any(v is None for v in [top_left, top_right, bottom_right,
+                                   bottom_left]):
+            raise ValueError('Could not get presets, make sure you set presets'
+                             ' first using the `set_presets` method.')
+        rows = m_rows or self.m_n_points[0]
+        columns = n_columns or self.m_n_points[1]
+
+        a_coeffs, b_coeffs = mesh_interpolation(top_left, top_right,
+                                                bottom_right, bottom_left)
+        self.coefficients = a_coeffs.tolist() + b_coeffs.tolist()
+        x_points, y_points = [], []
+
+        xx, yy = get_unit_meshgrid(m_rows=rows, n_columns=columns)
+
+        # return x_points, y_points
+        for rowx, rowy in zip(xx, yy):
+            for x, y in zip(rowx, rowy):
+                i, j = convert_to_physical(a_coeffs=a_coeffs,
+                                           b_coeffs=b_coeffs,
+                                           logic_x=x, logic_y=y)
+                x_points.append(i)
+                y_points.append(j)
+        if snake_like:
+            x_points = snake_grid_list(
+                np.array(x_points).reshape(rows, columns))
+        self.positions_x = x_points
+        self.positions_y = y_points
+        return x_points, y_points
+
+    def compute_mapped_point(self, sample_name, m_row, n_column,
```

Comment:
Just realized that this function also needs to know if it was saved in snake-like pattern .... maybe it was not a good idea to save the points in a snake-like pattern.
😭 

I should probably add one more little field in the file to indicate if snake-like

Reply:
I think this is not True actually.... i will come back to this.